src/Common/tests/System/Buffers/NativeMemoryManager.cs

```diff
@@ -54,8 +59,12 @@ public bool IsRetained
 
         public override unsafe MemoryHandle Pin(int elementIndex = 0)
         {
-            if (elementIndex < 0 || elementIndex > _length)
+            // Note that this intentionally allows elementIndex == _length to
+            // support pinning zero-length instances.
```

Comment:
And just to be clear, this isn't just the case where _length == 0.  It also occurs when the resulting `Memory<byte>` is sliced down to be 0-length, e.g.
```C#
using (var nmm = new NativeMemoryManager(5))
{
    Memory<byte> m = nmm.Memory;
    m = m.Slice(5);
    m.Pin().Dispose();
}
```
will end up calling this `Pin(int)` method with `elementIndex == 5`.